dali/util/nvml.h

```diff
@@ -252,6 +252,20 @@ inline bool HasCuda11NvmlFunctions() {
   return nvmlHasCuda11NvmlFunctions();
 }
 
+inline float GetDriverVersion() {
+  if (!nvmlIsInitialized()) {
+    return 0;
+  }
+
+  float driver_version = 0;
+  char version[NVML_SYSTEM_DRIVER_VERSION_BUFFER_SIZE];
+
+  CUDA_CALL(nvmlSystemGetDriverVersion(version, sizeof version));
```

Comment:
I suggest that we switch to CUDA Driver API for that.
```CUresult cuDriverGetVersion ( int* driverVersion )```
The documentation wasn't clear before CUDA 10, but in 10.2 it says that it
> Returns in *driverVersion the version of CUDA supported by the driver. The version is returned as (1000 major + 10 minor). For example, CUDA 9.2 would be represented by 9020.

Reply:
But the cuDriverGetVersion gives me CUDA version, not the driver we are using. Can we safely map those two things?

Reply:
I guess (in)compatibility mode could mess it up.

Reply:
Now that I think of it, most (all?) of the features we would query the driver for depend on user-mode features. As such, if NVML is not available, we could just conservatively report the oldest driver that supports the CUDA version reported by `cuDriverGetVersion`.